janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphDefaultSchemaMaker.java

```diff
@@ -34,6 +40,39 @@ public Cardinality defaultPropertyCardinality(String key) {
         return Cardinality.SINGLE;
     }
 
+    @Override
+    public PropertyKey makePropertyKey(PropertyKeyMaker factory, Object value) {
+        String name = factory.getName();
+        Class klass = null;
+        if (value instanceof String) {
+            klass = String.class;
+        } else if (value instanceof Character) {
+            klass = Character.class;
+        } else if (value instanceof Boolean) {
+            klass = Boolean.class;
+        } else if (value instanceof Byte) {
+            klass = Byte.class;
+        } else if (value instanceof Short) {
+            klass = Short.class;
+        } else if (value instanceof Integer) {
+            klass = Integer.class;
+        } else if (value instanceof Long) {
+            klass = Long.class;
+        } else if (value instanceof Float) {
+            klass = Float.class;
+        } else if (value instanceof Double) {
+            klass = Double.class;
```

Comment:
I'll fix this in another commit. There's _quite_ a few places that mention `Precision` and `Decimal`. They've been out of Titan for a while though.